### PR TITLE
feat: configurable initial relay retries; async db read helpers

### LINKED ISSUE #411: coproc: make relay retries configurable
We need to make this value configurable via CLI 

https://github.com/InfinityVM/InfinityVM/blob/d42e4bbe2a809b985a6d5d5d29cab2ab1f7a2a36/crates/coproc/src/relayer.rs#L40

Once it is configurable, we will want to increase this value to a very high number in the testnet.

### PR DESCRIPTION
**Pull request by emostov**
Changes
- Rename `max_retries` to `dlq_max_retries`. 
- Introduce `initial_relay_max_retries`. We set a very high high default of 1,000 because for the initial exchange use case we cannot proceed with relaying the next job until the current job makes its way on chain. Closes #411 
- Add async methods for db reads that use spawn blocking to perform the reads. This isn't related to the new configuration param, but @Avi-D-coder noticed that we where doing blocking DB reads within tokio async code which is not ideal so figure I would quickly add the change here. The naming convention I used here is that I appended `_sync` to the existing functions, and left the new async version names the original names

### CODE REVIEW

File: crates/coproc/src/relayer.rs [hunk 1]
```diff
@@ -37,7 +37,8 @@ use tracing::{error, info};
 
 /// Delay between retrying failed jobs, in milliseconds.
 const JOB_RETRY_DELAY_MS: u64 = 500;
-const JOB_RETRY_COUNT: usize = 3;
+/// Max duration between retries in `relay_job_result`.
+const JOB_RETRY_MAX_DELAY_MS: u64 = 30 * 1_000;
```
Comment (emostov) [inline crates/coproc/src/relayer.rs:41]: Semi arbitrary. Open to input here

**Comment by emostov** [comment]
The failing test looks to be a persistent issue. Investigating

File: crates/coproc/src/cli.rs [hunk 2]
```diff
@@ -124,9 +124,15 @@ struct Opts {
...
+    /// Max number of retries for relaying a job in the dead letter queue (DLQ) before giving up.
+    /// Note that each job in the DLQ is tried once before trying again.
     #[arg(long, default_value_t = 3)]
-    max_retries: usize,
+    dlq_max_retries: usize,
+
+    /// Max number of retries when initially attempting to relay a job. If a job fails this many
+    /// times it will be added to the DLQ.
+    #[arg(long, default_value_t = 1_000)]
+    initial_relay_max_retries: usize,
```
Comment (Avi-D-coder) [inline crates/coproc/src/cli.rs:135]: Tiny NIT: if we are casting to u32 why not just make the config values u32

**Review by Avi-D-coder — APPROVED** [review]
LGTM once the CI passes.

**Comment by lucasege** [comment]
Is there an issue for this? perhaps https://github.com/InfinityVM/InfinityVM/issues/410

**Pull request merged**